website/docs/cli/commands/test.mdx

```diff
@@ -1,12 +1,165 @@
+The option can be used multiple times to specify several files.
+
+* `-json` Set for test outputs to be formatted as JSON.
+
+* `-no-color` Disable color codes in the command output.
+
+* `-verbose` Print the plan or state for each test run block as it executes.
+
+## Usage examples
+
+### Flat structure
+
+**GIVEN** the following OpenTofu module structure:
+
+```commandline
+.
+├── LICENSE
+├── README.md
+├── main.tf
+├── main.tftest.hcl
+├── foo.tf
+├── foo.tftest.hcl
+├── bar.tf
+├── bar.tftest.hcl
+├── outputs.tf
+└── variables.tf
+```
+
+**WHEN** `tofu test` is executed in the project directory,
+
+**THEN** the tests defined in all three files, `main.tftest.hcl`, `foo.tftest.hcl` and `bar.tftest.hcl`, will be run.
+
+**WHEN** `tofu test -filter=foo.tftest.hcl -filter=bar.tftest.hcl` is executed in the project directory,
+
+**THEN** the tests defined in the files `foo.tftest.hcl` and `bar.tftest.hcl` will be run only.
+
+### Nested structure
+
+**GIVEN** the following OpenTofu module structure:
+
+```commandline
+.
+├── LICENSE
+├── README.md
+├── foo.tf
+├── foo.tftest.hcl
+├── bar.tf
+├── bar.tftest.hcl
+├── main.tf
+├── main.tftest.hcl
+├── tests
+│   ├── foo.tftest.hcl
+│   └── bar.tftest.hcl
+├── modules
+│   ├── module_foo
+│   │   ├── LICENSE
+│   │   ├── README.md
+│   │   ├── foo.tf
+│   │   ├── foo.tftest.hcl
+│   │   ├── main.tf
+│   │   ├── main.tftest.hcl
+│   │   ├── outputs.tf
+│   │   └── variables.tf
+│   └── module_bar
+│       ├── LICENSE
+│       ├── README.md
+│       ├── foo.tf
+│       ├── foo.tftest.hcl
+│       ├── bar.tf
+│       ├── bar.tftest.hcl
+│       ├── main.tf
+│       ├── main.tftest.hcl
+│       ├── outputs.tf
+│       └── variables.tf
+├── outputs.tf
+└── variables.tf
+```
+
+**WHEN** `tofu test` is executed in the project root directory,
+
+**THEN** the tests defined in the following files will be run (note that the files are sorted in the execution order):
+
+- bar.tftest.hcl,
+- foo.tftest.hcl,
+- main.tftest.hcl,
+- tests/bar.tftest.hcl,
+- tests/foo.tftest.hcl.
+
+**WHEN** `tofu test -test-directory=modules/module_bar` is executed in the project root directory,
+
+**THEN** the tests defined in the following files will be run (note that the files are sorted in the execution order):
+
+- bar.tftest.hcl,
+- foo.tftest.hcl,
+- main.tftest.hcl,
+- modules/module_bar/bar.tftest.hcl,
+- modules/module_bar/foo.tftest.hcl,
+- modules/module_bar/main.tftest.hcl.
+
+**WHEN** `tofu test -test-directory=modules/module_bar -filter=bar.tftest.hcl` is executed in the project root directory,
+
+**THEN** the tests defined in the file `bar.tftest.hcl` will be run only.
+
+**WHEN** `tofu test -test-directory=modules/module_bar -filter=modules/module_bar/bar.tftest.hcl` is executed in the project root directory,
+
+**THEN** the tests defined in the file `modules/module_bar/bar.tftest.hcl` will be run only.
```

Comment:
I understand we want to be comprehensive, but this is a very complex example, could we simplify this a bit more? I don't think we need foo/bar in every directory maybe?

Reply:
Thanks for idea - I adjusted the docu accordingly.